#171784087 Setup the API for User Sign Out

fordarnold opened the pull request:
#### What does this PR do?
This PR implements the unit tests and logic for User Sign Out API. Users should be able to log out of the application.

#### Description of Task to be completed?
- write unit test cases for user logout API
- setup controllers and middleware
- set default role for new User

#### How should this be manually tested?
> First, you must be signed in as a User.

- Logout a User by sending an HTTP POST request to the API endpoint `/api/v1/users/logout`, with the request header `Authorization` containing a valid token of the User you want to logout.

[See screenshots below for more details...]

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[171784087](https://www.pivotaltracker.com/story/show/171784087)

#### Screenshots

##### Logout a User
![Logout a User](https://user-images.githubusercontent.com/1271441/79696844-6bd80c80-827f-11ea-9567-b0c6b59c31e6.png)

##### Attempt to Logout a User with Invalid Token
![Attempt to Logout a User with Invalid Token](https://user-images.githubusercontent.com/1271441/79696850-70042a00-827f-11ea-8f33-d82e0783877c.png)

##### Attempt to Logout a User without Token
![Attempt to Logout a User without Token](https://user-images.githubusercontent.com/1271441/79696852-72ff1a80-827f-11ea-8b9d-d09409cabf23.png)

irfiacre commented on package.json:
```diff
@@ -13,18 +13,19 @@
-    "start": "npm run migrationSeeds && babel-node -- server/index.js",
-    "start:dev": "nodemon --exec babel-node ./server/index.js",
+    "start": "npm run migration && npm run migration:seed && babel-node -- server/index.js",
+    "start:dev": "npm run migration:reset && nodemon --exec babel-node ./server/index.js",
```
I think the `start:dev` can remain as it was because this **npm run migration:reset** would not work for windows users, as you remember.

fordarnold replied:
@irfiacre, I added separate scripts for Windows machines... You will see them in the `package.json`... I think just use these instead.

```
"win:start:dev": "nodemon --exec babel-node ./server/index.js",
"win:createTables": "npx sequelize-cli db:migrate",
"win:drop": "npx sequelize-cli db:migrate:undo:all",
"win:devtable": " npm run drop && npm run createTables && npx sequelize-cli db:seed:all"
```

irfiacre replied:
@fordarnold  but since `start:dev` was working for all of us and I guess on `start:dev` migrations are not that necessary, I think It's better to keep one script!

fordarnold replied:
I think it would be better to have both since I'm running migrations on my machine when using `start:dev` script (It makes it easier for me since I don't have to run migrations separately, and I just do it all in one script)...

Maybe we can first hear what @bdushimi and @Francois-MUGOROZI have to say about it.

irfiacre replied:
I understand now.

fordarnold replied:
@irfiacre The other option would be to leave `start:dev` as it was originally, and then instead create another script: ` "unix:start:dev": "npm run migration:reset && nodemon --exec babel-node ./server/index.js" ` which would be used by those with Unix machines.

So I believe we would have to agree as a team which options would be better...

irfiacre commented on server/services/RoleService.js:
```diff
@@ -0,0 +1,44 @@
+    });
+  }
+
+  // static truncateRows() {
```
If we don't need this you could simply just delete it.

fordarnold committed:
feature(auth): implement User Logout API feature

Main requirements for completing this task...

- write unit test cases for user logout API
- setup controllers and middleware
- set default role for new User
- document the API with Swagger

Other minor updates included...

- refactor(users): update user info encrypted in access token

[Finishes #171784087]

kabundege reviewed (commented):
- squash you commits
- solve your conflicts
> Why didn't you use headers or local-storage in signout, it would have been easier. 
> **My question is how does your Method work in a browser?**

fordarnold commented:
> * squash you commits
> * solve your conflicts
> 
> > Why didn't you use headers or local-storage in signout, it would have been easier.
> > **My question is how does your Method work in a browser?**

In the browser, you need to pass a valid Access token in the HTTP request headers.

irfiacre commented on server/controllers/usercontroller.js:
```diff
@@ -125,59 +120,64 @@ class User {
       user: userInfo,
     };
 
-    const message = 'You have signed in successfully';
```
Upon signing in, where do you pass the token into the database?

irfiacre reviewed (commented):
Good job on the work you did.
